agent/ecscni/plugin.go

```diff
@@ -69,7 +70,24 @@ func NewClient(cfg *Config) CNIClient {
 
 // SetupNS will set up the namespace of container, including create the bridge
 // and the veth pair, move the eni to container namespace, setup the routes
-func (client *cniClient) SetupNS(cfg *Config) (*current.Result, error) {
+func (client *cniClient) SetupNS(cfg *Config, ctx context.Context) (*current.Result, error) {
+	var result *current.Result
+	var err error
+	setupDone := make(chan struct{})
+	go func() {
+		result, err = client.setupNS(cfg)
+		setupDone <- struct{}{}
+	}()
+
+	select {
+	case <-ctx.Done():
+		return nil, errors.Errorf("cni setup: setup the container namespace timedout")
```

Comment:
this can just be `errors.New`? 
also, message can be `"cni: container namespace setup timed out"`